### ISSUE TITLE
Improve flakyness of tests

### ISSUE THREAD

**Issue by khendrikse** [netlify/cli#5265]
### Which problem is this feature request solving?

A lot of tests on PR's are flaky. This causes devs to have to rerun tests. 

It's usually time-outs that cause the tests to fail.

### Describe the solution you'd like

During a call we spoke about looking at the following:

* what tests are most likely to fail
* restructuring (some people had ideas)
* trying out Vitest

### Pull request (optional)

- [X] I can submit a pull request.

**Comment by danez** [comment]
Before we can use vitest, we need to migrate all files to ESM.

On thing I had on my mind is the writing of the global config file here. This happens for every test and leads to the tests on windows fail with EPERM error when multiple tests write at the same time I think. So maybe we can disable this in tests or mock the filesystem for these paths or write to a unique location for the test, like a tmp folder.

**Comment by danez** [comment]
The ESM migration is pretty much done, there is this PR: #5298 and after that I already have another PR ready.

I had also more ideas while migrating and getting more insights:

- same as mentioned above: Mock get-global-config in tests, so that we do not even read the config file, as well as write it.
- Every `netlify dev` test does spin up a dev server and runs tests against it. We could optimize and group tests that use the same fixtures. Then only spin up one dev-server for all of these tests and run all the tests against this one dev server. This will also probably reduce the test time by a lot.
